src/coreclr/src/ilasm/grammar_after.cpp

```diff
@@ -1520,19 +1505,15 @@ void AsmParse::ParseFile(ReadStream* stream)
 char* AsmParse::fillBuff(__in_opt __nullterminated char* pos)
 {
     int iPutToBuffer;
-    int iOptions = IS_TEXT_UNICODE_UNICODE_MASK;
     g_uCodePage = CP_ACP;
     iPutToBuffer = (int)penv->in->getAll(&(penv->curPos));
 
     penv->endPos = penv->curPos + iPutToBuffer;
     if(iPutToBuffer > 128) iPutToBuffer = 128;
-    if(IsItUnicode(penv->curPos,iPutToBuffer,&iOptions))
+    if(iPutToBuffer >= 4 && (penv->curPos[0] & 0xFF) == 0xFF && (penv->curPos[1] & 0xFF) == 0xFE && ((penv->curPos[2] & 0xFF) == 0x00 || (penv->curPos[3] & 0xFF) != 0x00))
```

Comment:
`(penv->curPos[3] & 0xFF) != 0x00)` case was incorrectly using UTF16 before and it will incorrectly use UTF8 now.  Should this error out instead?